### PR TITLE
Add amp-form test for pre-existing redirection behavior

### PR DESCRIPTION
**Pull request by caroqliu**
Follow up to #25473 to prevent future breakage

### CODE REVIEW

File: extensions/amp-form/0.1/test/test-amp-form.js [hunk 1]
```diff
@@ -2426,6 +2426,28 @@ describes.repeated(
                 });
               });
 
+              it('should redirect users if header is set but json rejects', () => {
+                sandbox.stub(ampForm.xhr_, 'fetch').returns(
+                  Promise.resolve({
```
Comment (wassgha) [inline extensions/amp-form/0.1/test/test-amp-form.js]: Let's use `.resolves` here

Comment (caroqliu) [reply extensions/amp-form/0.1/test/test-amp-form.js]: Would prefer to keep as is since that is more self-consistent with other stubs in this file. WDYT?

Comment (wassgha) [reply extensions/amp-form/0.1/test/test-amp-form.js]: (y) we can switch them on a different PR since this is a large file

Comment (caroqliu) [reply extensions/amp-form/0.1/test/test-amp-form.js]: Follow up PR: #25539

**Pull request merged**